go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -32,6 +48,818 @@ func (c *Controller) processNextANPWorkItem(wg *sync.WaitGroup) bool {
+			asIndex := GetANPPeerAddrSetDbIDs(desiredANPState.name, rule.gressPrefix, fmt.Sprintf("%d", rule.priority),
+				controllerName, libovsdbops.AddressSetAdminNetworkPolicy)
+			addrSet, err := c.addressSetFactory.EnsureAddressSet(asIndex)
+			if err != nil {
+				return err
+			}
+			rule.addrSet = addrSet
+			rule.addrSet.SetIPs(stringsToIPs(rule.podIPs.UnsortedList()))
+		}
+		// now recreate the address-sets and then the acls
+		ops, err = libovsdbops.CreateOrUpdateACLsOps(c.nbClient, ops, desiredACLs...)
+		if err != nil {
+			return fmt.Errorf("failed to create new ACL ops: %v", err)
+		}
+		ops, err = libovsdbops.AddACLsToPortGroupOps(c.nbClient, ops, portGroupName, desiredACLs...)
+		if err != nil {
+			return fmt.Errorf("failed to create ops to add port to a port group: %v", err)
+		}
+		fullPeerRecompute = true
+	}
+
+	// Did ANP.Spec.Ingress rules get updated?
+	// (at this stage the length of ANP.Spec.Ingress hasn't changed, so individual rules either got updated at their values or positions are switched)
+	// The fields that we care about for rebuilding ACLs are
+	// (i) `ports` (ii) `actions` (iii) priority for a given rule
+	// The changes to peer labels, peer pod label updates, namespace label updates etc can be inferred
+	// from the podIPs cache we store.
+	// Did the ANP.Spec.Ingress.Peers Change?
+	// 1) ANP.Spec.Ingress.Peers.Namespaces changed && ||
+	// 2) ANP.Spec.Ingress.Peers.Pods changed && ||
+	// 3) A namespace started or stopped matching the peer && ||
+	// 4) A pod started or stopped matching the peer
+	// If yes we need to recompute the IPs present in our ANP's peer's address-sets
+	atLeastOneRuleUpdated := false
+	if len(desiredANPState.ingressRules) == len(currentANPState.ingressRules) {
+		for i := range desiredANPState.ingressRules {
+			desiredRule := desiredANPState.ingressRules[i]
+			currentRule := currentANPState.ingressRules[i]
+			ipsToAdd := desiredRule.podIPs.Difference(currentRule.podIPs)
+			klog.Infof("Adding podIPs %+v to address-set %s for ANP %s", ipsToAdd, desiredRule.addrSet.GetName(), desiredANPState.name)
+			addrOps, err := desiredRule.addrSet.AddIPsReturnOps(stringsToIPs(ipsToAdd.UnsortedList()))
+			if err != nil {
+				return fmt.Errorf("unable to construct address-set %s's ip add ops: %v", desiredRule.addrSet.GetName(), err)
+			}
+			ops = append(ops, addrOps...)
+			ipsToRemove := currentRule.podIPs.Difference(desiredRule.podIPs)
+			klog.Infof("Deleting podIPs %+v from address-set %s for ANP %s", ipsToRemove, desiredRule.addrSet.GetName(), desiredANPState.name)
+			addrOps, err = desiredRule.addrSet.DeleteIPsReturnOps(stringsToIPs(ipsToRemove.UnsortedList()))
+			if err != nil {
+				return fmt.Errorf("unable to construct address-set %s's ip delete ops: %v", desiredRule.addrSet.GetName(), err)
+			}
+			ops = append(ops, addrOps...)
+			if currentRule.action != desiredRule.action || !reflect.DeepEqual(desiredRule.ports, currentRule.ports) {
```

Comment:
hmm not sure I follow this comment because whether action and ports are different or not, we will need to still go through the peer changes always i guess...right?